tests/pages/job.js

```diff
@@ -27,6 +27,23 @@ export default create({
     avatarSrc: attribute('src', 'img'),
   },
 
+  hasWaitingStates: isVisible('.job-waiting-stages-container'),
+  stageOneLoading: hasClass('spinner', '.job-waiting-stages-container .stage-1-container svg'),
+  stageTwoLoading: hasClass('spinner', '.job-waiting-stages-container .stage-2-container svg'),
+  stageThreeLoading: hasClass('spinner', '.job-waiting-stages-container .stage-3-container svg'),
+  stageOneLoaded: hasClass('stage-loaded-ring', '.job-waiting-stages-container .stage-1-container'),
+  stageTwoLoaded: hasClass('stage-loaded-ring', '.job-waiting-stages-container .stage-2-container'),
+  stageThreeLoaded: hasClass('stage-loaded-ring', '.job-waiting-stages-container .stage-3-container'),
+  stageOneWaitingText: text('.job-waiting-stages-container .stage-1-container span.waiting-stage-1-text'),
+  stageTwoWaitingText: text('.job-waiting-stages-container .stage-2-container .waiting-stage-2-text'),
+  stageThreeWaitingText: text('.job-waiting-stages-container .stage-3-container .waiting-stage-3-text'),
+  stageOneText: text('.job-waiting-stages-container .stage-1-message'),
+  stageTwoText: text('.job-waiting-stages-container .stage-2-message'),
+  stageThreeText: text('.job-waiting-stages-container .stage-3-message'),
+  stageOneLoadingLineInactive: hasClass('loading-line-1', '.loading-line-container-1 span'),
+  stageTwoLoadingLineInactive: hasClass('loading-line-2', '.loading-line-container-2 span'),
+  stageOneLoadingLineActive: hasClass('loading-line', '.loading-line-container-1 span'),
+  stageTwoLoadingLineActive: hasClass('loading-line', '.loading-line-container-2 span'),
```

Comment:
I think it could lead to more clear and expressive test assertions to structure this using what the page object library calls [components](http://ember-cli-page-object.js.org/docs/v1.15.x/components). It could look something like this:

```
waitingStates: {
  scope: '.job-waiting-states-container',

  one: {
    scope: '.stage-1-container',
    isInactive: hasClass('not-loaded-ring'),
    isLoading: hasClass('spinner', 'svg'),
    isLoaded: hasClass('stage-loaded-ring'),
  },
  
  firstLine: {
    scope: '.loading-line-container-1 span',
    isInactive: notHasClass('loading-line'),
    isActive: hasClass('loading-line'),
  },

…
}
```

Then the assertions can be more concise and descriptive:

```
# created

assert.ok(jobPage.waitingStates.isVisible);

assert.ok(jobPage.waitingStates.one.isLoading);
assert.ok(jobPage.waitingStates.firstLine.isInactive);
assert.ok(jobPage.waitingStates.two.isInactive);
assert.ok(jobPage.waitingStates.secondLine.isInactive);
assert.ok(jobPage.waitingStates.three.isInactive);

# change state to queued

assert.ok(jobPage.waitingStates.one.isLoaded);
assert.ok(jobPage.waitingStates.firstLine.isActive);
assert.ok(jobPage.waitingStates.two.isLoading);
…

# change state to started

assert.ok(jobPage.waitingStates.isHidden);
```

Does that make sense?
    
I avoided the name “stage” here because it’s confusing to me to have two different concepts being called stages.

Reply:
also I see some `show-stage-message` and `hide-stage-message` classes that have no corresponding assertions, I guess for mobile? Those could be included here too, and then in the assertions.

Reply:
I think these can be deleted now that they’re covered by the more structured objects?